sql/parser/builtins.go

```diff
@@ -575,3 +610,13 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+func abs(x DInt) (DInt, error) {
+	switch {
+	case x == math.MinInt64:
+		return x, errAbsOfMinInt64
```

Comment:
I don't think we should introduce `DUint` as that would bring along with it all sorts of complexity in handling unsigned vs signed operations. Agreed that the error message should not mention a Go type. Perhaps something like: `abs of min integer value (-9223372036854775808) not defined`